src/status_im/infra/market/transform.cljs

```diff
@@ -0,0 +1,70 @@
+(ns status-im.infra.market.transform
+  (:require
+    [camel-snake-kebab.extras :as cske]
+    [utils.transforms :as transforms]))
+
+(defn json->clj-kebab-case
+  [json]
+  (->> json
+       transforms/json->clj
+       (cske/transform-keys transforms/->kebab-case-keyword)))
+
+(defn signal-payload-json->clj
+  [signal-payload]
+  (->> signal-payload
+       transforms/js->clj
+       :message
+       json->clj-kebab-case))
+
+(defn leaderboard-message->positions-by-token
+  "Extract token ids and index them by position in a leaderboard"
+  [{:keys [page page-size cryptocurrencies] :as _leaderboard-message}]
+  (let [start-from (-> page
+                       dec
+                       (* page-size)
+                       inc)]
+    (reduce (fn [acc cryptocurrency]
+              (let [position (+ start-from (count acc))]
+                (assoc acc (:id cryptocurrency) position)))
+            {}
+            cryptocurrencies)))
+
+(defn leaderboard-message->prices-by-token-id
+  "Extract only token ids and preserve order"
+  [{:keys [cryptocurrencies]}]
+  (reduce
+   (fn [acc token-data]
+     (assoc acc
+            (:id token-data)
+            (select-keys token-data
+                         [:id
+                          :current-price
+                          :market-cap
+                          :total-volume
+                          :price-change-percentage-24h])))
+   {}
+   cryptocurrencies))
+
+(defn price-update-message->price-updates-by-token-id
+  "Extract only token ids and preserve order"
+  [{:keys [prices]}]
+  (reduce
+   (fn [acc {:keys [id] :as price}]
+     (assoc acc id price))
+   {}
+   prices))
+
+(defn leaderboard-message->token-data-by-token-id
+  "Extract only token ids and preserve order"
+  [{:keys [cryptocurrencies]}]
+  (reduce
+   (fn [acc token-data]
+     (assoc acc
+            (:id token-data)
+            (select-keys token-data
```

Comment:
I would avoid `select-keys` inside loops to remove the key iteration cost on every instance of `token-data`.

Reply:
Do you suggest using destructuring instead?